test/src/main/java/org/axonframework/test/aggregate/ResultValidatorImpl.java

```diff
@@ -92,12 +102,65 @@ public ResultValidator expectSuccessfulHandlerExecution() {
         return expectReturnValueMatching(anything());
     }
 
+    @Override
+    public ResultValidator expectScheduledDeadlineMatching(Duration duration,
+                                                           Matcher<? super DeadlineMessage<?>> matcher) {
+        deadlineManagerValidator.assertScheduledDeadlineMatching(duration, matcher);
+        return this;
+    }
+
+    @Override
+    public ResultValidator expectScheduledDeadline(Duration duration, Object deadline) {
+        return expectScheduledDeadlineMatching(duration, messageWithPayload(Matchers.equalTo(deadline, fieldFilter)));
+    }
+
+    @Override
+    public ResultValidator expectScheduledDeadlineOfType(Duration duration, Class<?> deadlineType) {
+        return expectScheduledDeadlineMatching(duration, messageWithPayload(any(deadlineType)));
+    }
+
+    @Override
+    public ResultValidator expectScheduledDeadlineMatching(Instant scheduledTime,
+                                                           Matcher<? super DeadlineMessage<?>> matcher) {
+        deadlineManagerValidator.assertScheduledDeadlineMatching(scheduledTime, matcher);
+        return this;
+    }
+
+    @Override
+    public ResultValidator expectScheduledDeadline(Instant scheduledTime, Object deadline) {
+        return expectScheduledDeadlineMatching(scheduledTime,
+                                               messageWithPayload(Matchers.equalTo(deadline, fieldFilter)));
+    }
+
+    @Override
+    public ResultValidator expectScheduledDeadlineOfType(Instant scheduledTime, Class<?> deadlineType) {
+        return expectScheduledDeadlineMatching(scheduledTime, messageWithPayload(any(deadlineType)));
+    }
+
+    @Override
+    public ResultValidator expectNoScheduledDeadlines() {
+        deadlineManagerValidator.assertNoScheduledDeadlines();
+        return this;
+    }
+
+    @Override
+    public ResultValidator expectDeadlinesMetMatching(Matcher<? extends List<? super DeadlineMessage<?>>> matcher) {
+        deadlineManagerValidator.assertDeadlinesMetMatching(matcher);
+        return this;
+    }
+
+    @Override
+    public ResultValidator expectDeadlinesMet(Object... expected) {
+        deadlineManagerValidator.assertDeadlinesMet(expected);
+        return this;
+    }
+
     @Override
     public ResultValidator expectReturnValue(Object expectedReturnValue) {
         if (expectedReturnValue == null) {
             return expectReturnValueMatching(nullValue());
         }
-        return expectReturnValueMatching(equalTo(expectedReturnValue));
+        return expectReturnValueMatching(CoreMatchers.equalTo(expectedReturnValue));
```

Comment:
What's the idea behind using `CoreMatchers` here?

Reply:
I see this is already adjusted in later commits, please disregard these comments.